{
  throw new RuntimeException(function + " failed: " + error);
}
